private/pkg/observability/observability.go

```diff
@@ -79,6 +86,11 @@ func Start(options ...StartOption) io.Closer {
 		view.RegisterExporter(traceViewExportCloser)
 		closers = append(closers, traceViewExportCloser)
 	}
+	for _, tracerProviderCloser := range startOptions.tracerProviderClosers {
+		tracerProviderCloser := tracerProviderCloser
+		otel.SetTracerProvider(tracerProviderCloser)
```

Comment:
This sets the trace provider as the globally registered one - it feels like assigning this in a loop could indicate a misconfiguration if `len(tracerProviderClosers) > 1`.